cmd/nerdctl/system_prune_linux_test.go

```diff
@@ -51,4 +59,30 @@ func TestSystemPrune(t *testing.T) {
 	base.Cmd("ps", "-a").AssertNoOut(tID)
 	base.Cmd("network", "ls").AssertNoOut(nID)
 	base.Cmd("images").AssertNoOut("alpine")
+
+	buildctlBinary, err := buildkitutil.BuildctlBinary()
+	if err != nil {
+		t.Fatal(err)
+	}
+	host, err := buildkitutil.GetBuildkitHost(testutil.Namespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	buildctlArgs := buildkitutil.BuildctlBaseArgs(host)
+	buildctlArgs = append(buildctlArgs, "prune")
+	logrus.Debugf("running %s %v", buildctlBinary, buildctlArgs)
+	buildctlCmd := exec.Command(buildctlBinary, buildctlArgs...)
```

Comment:
We should test nerdctl  rather than buildctl.

Reply:
https://github.com/containerd/nerdctl/actions/runs/3148979445/jobs/5120090767#step:6:884

```
=== RUN   TestSystemPrune
    system_prune_linux_test.go:65: exec: "buildctl": executable file not found in $PATH
--- FAIL: TestSystemPrune (3.80s)
```

Reply:
> We should test nerdctl rather than buildctl.

Oh, I am not testing `buildctl`. It would execute `system prune -f --volumes --all` first, then make an assertion on the output `buildctl cache prune` should be `Total: 0B`, for making sure that all cache has been cleared.

And it requires `buildkitd` service started, but it would not be compatible with rootless. Maybe I should skip this testcase under rootless?

Reply:
> Oh, I am not testing buildctl. It would execute system prune -f --volumes --all first, then make an assertion on the output buildctl cache prune should be Total: 0B, for making sure that all cache has been cleared.

Maybe you can use `buildctl du`.

> And it requires buildkitd service started, but it would not be compatible with rootless. Maybe I should skip this testcase under rootless?

For rootless, `TestSystemPrune` should run  but checking the cache size can be skipped. We'll need `nerdctl system df` with making it report the build cache size and we can use this for rootless `TestSystemPrune` test.

Reply:
>  it would not be compatible with rootless

Why not compatible?